### PR TITLE
[SPARK-29889][SQL][test] unify the interval tests

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

move interval tests to `interval.sql`, and import it to `ansi/interval.sql`

### Why are the changes needed?

improve test coverage

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @maropu @MaxGekk @yaooqinn

File: sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql [hunk 1]
```diff
@@ -1,187 +1,15 @@
...
-  '10 9:8:7.987654321' day to second,
-  '10' day,
-  '11' hour,
-  '12' minute,
-  '13' second,
-  '13.123456789' second;
-
-select map(1, interval 1 day, 2, interval 3 week);
-
-select map(1, 1 day, 2, 3 week);
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql]: tests above are already covered by `interval.sql`

File: sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql [hunk 2]
```diff
@@ -1,187 +1,15 @@
...
-
--- Interval year-month arithmetic
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/ansi/interval.sql]: tests below are moved to `interval.sql`

File: sql/core/src/test/resources/sql-tests/inputs/group-by.sql [hunk 3]
```diff
@@ -161,67 +161,3 @@ SELECT * FROM (SELECT COUNT(*) AS cnt FROM test_agg) WHERE cnt > 1L;
 SELECT count(*) FROM test_agg WHERE count(*) > 1L;
 SELECT count(*) FROM test_agg WHERE count(*) + 1L > 1L;
 SELECT count(*) FROM test_agg WHERE k = 1 or k = 2 or count(*) + 1L > 1L or max(k) > 1;
-
--- sum interval values
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/group-by.sql]: moved to `interval.sql`

File: sql/core/src/test/resources/sql-tests/inputs/interval.sql [hunk 4]
```diff
@@ -128,6 +131,179 @@ select interval 1 year '2-1' year to month;
...
 select interval '12:11:10' hour to second '1' year;
+select interval (-30) day;
+select interval (a + 1) day;
+select interval 30 day day day;
 
--- awareness of the positive sign before interval type
-select +interval '1 second';
+-- sum interval values
+-- null
+select sum(cast(v as interval)) from VALUES ('1 seconds'), ('2 seconds'), (null) t(v) where v is null;
```
Comment (yaooqinn) [inline sql/core/src/test/resources/sql-tests/inputs/interval.sql]: select sum(cast(v as interval)) from VALUES (null) t(v);

**Comment by yaooqinn** [comment]
https://github.com/apache/spark/blob/f926809a1f6b3dd6041518a98b115e42d9692836/sql/core/src/test/resources/sql-tests/inputs/postgreSQL/interval.sql#L330
these tests should be enabled

**Comment by cloud-fan** [comment]
do we support `make_interval(years := 2)`?

**Comment by cloud-fan** [comment]
and pgsql tests are ported from pgsql, I don't think we should import spark tests in pgsql tests.

**Comment by yaooqinn** [comment]
> do we support `make_interval(years := 2)`?

not supported yet, but the ticket is resolved as done

**Comment by MaxGekk** [comment]
>> do we support make_interval(years := 2)?

> not supported yet, but the ticket is resolved as done

The ticket aimed to support `make_interval`. Passing parameters by names is independent big feature. Need a separate ticket for it.

**Review by maropu — APPROVED** [review]
This clean-up looks nice! LGTM

**Review by HyukjinKwon — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks for review, merging to master!

**Pull request closed without merging**